qadence/ml_tools/train_grad.py

```diff
@@ -122,9 +125,17 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
         model = model.module.to(device=device, dtype=dtype)
     else:
         model = model.to(device=device, dtype=dtype)
-    # initialize tensorboard
-    writer = SummaryWriter(config.folder, purge_step=init_iter)
+    # initialize tracking tool
+    if config.tracking_tool == ExperimentTrackingTool.TENSORBOARD:
+        writer = SummaryWriter(config.folder, purge_step=init_iter)
+    else:
+        mlflowconfig = MLFlowConfig()  # Set up credentials for mlflow tracking
+        writer = importlib.import_module("mflow")
```

Comment:
Not sure about others but I personally prefer to do a 
```
try:
    import X
except ImportError
     X = None
     logger.debug("X not available")
```
At the top of the file together with normal imports

Reply:
you dont always wanna import mlflow, only in case the user passes it in the trainconfig. generally, i agree with you but in this case it avoids an unnecessary import